subgraphs/playback/src/utils/server.ts

```diff
@@ -0,0 +1,181 @@
+import { readFileSync } from 'fs';
+import gql from 'graphql-tag';
+import { buildSubgraphSchema } from '@apollo/subgraph';
+import { ApolloServer, ApolloServerPlugin } from '@apollo/server';
+import { ApolloServerPluginSubscriptionCallback } from '@apollo/server/plugin/subscriptionCallback';
+import {
+  ApolloServerPluginLandingPageProductionDefault,
+  ApolloServerPluginLandingPageLocalDefault,
+} from '@apollo/server/plugin/landingPage/default';
+import resolvers from '../resolvers';
+import { ContextValue } from '../types/ContextValue';
+const port = process.env.PORT ?? '4002';
+const routerSecret = process.env.ROUTER_SECRET;
+
+import { WebSocketServer } from 'ws';
+import { useServer } from 'graphql-ws/lib/use/ws';
+import { ApolloServerPluginDrainHttpServer } from '@apollo/server/plugin/drainHttpServer';
+
+import express from 'express';
+import http from 'http';
+import SpotifyAPI from '../dataSources/spotify';
+import { GraphQLError } from 'graphql';
+import { MockedSpotifyDataSource, addUser } from '../utils/mocks';
+import logger from '../logger';
+import * as Sentry from '@sentry/node';
+
+export const app = express();
+export const httpServer = http.createServer(app);
+
+let typeDefs = gql(
+  readFileSync('schema.graphql', {
+    encoding: 'utf-8',
+  })
+);
+const schema = buildSubgraphSchema({
+  typeDefs,
+  resolvers,
+});
+const wsServer = new WebSocketServer({
+  server: httpServer,
+  path: '/ws',
+});
+const serverCleanup = useServer(
+  {
+    schema,
+    onConnect: (ctx) => {
+      if (ctx.connectionParams?.['authorization']) return true;
+      if (ctx.extra.request.headers?.['authorization']) return true;
+
+      //If there is no authorization, we will mock everything and use the referer for the users identifier in local mocked data
+      return true;
+    },
+    context: (ctx) => {
+      const routerAuthorization =
+        (ctx.connectionParams?.['authorization'] as string) ??
+        (ctx.extra.request.headers?.['authorization'] as string) ??
+        '';
+      checkRouterSecret(routerAuthorization);
+      const token =
+        (ctx.connectionParams?.['authorization'] as string) ??
+        (ctx.extra.request.headers?.['authorization'] as string);
+
+      const defaultCountryCode =
+        (ctx.connectionParams?.['country-code'] as string) ??
```

Comment:
TIL `X-` prefixed headers are not recommended 😄 https://stackoverflow.com/questions/3561381/custom-http-headers-naming-conventions

Reply:
#96 to track